x/subscription/keeper/cu_tracker.go

```diff
@@ -228,26 +230,41 @@ func (k Keeper) RewardAndResetCuTracker(ctx sdk.Context, cuTrackerTimerKeyBytes
 			)
 		} else {
 			utils.LogLavaEvent(ctx, k.Logger(ctx), types.MonthlyCuTrackerProviderRewardEventName, map[string]string{
-				"provider":       provider,
-				"sub":            sub,
-				"plan":           plan.Index,
-				"tracked_cu":     strconv.FormatUint(trackedCu, 10),
-				"plan_price":     plan.Price.String(),
-				"reward":         providerReward.String(),
-				"block":          strconv.FormatInt(ctx.BlockHeight(), 10),
-				"adjustment_raw": providerAdjustment.String(),
+				"provider":         provider,
+				"sub":              sub,
+				"plan":             subObj.PlanIndex,
+				"tracked_cu":       strconv.FormatUint(trackedCu, 10),
+				"credit_used":      creditToSub.String(),
+				"credit_remaining": subObj.Credit.String(),
+				"reward":           providerReward.String(),
+				"block":            strconv.FormatInt(ctx.BlockHeight(), 10),
+				"adjustment_raw":   providerAdjustment.String(),
 			}, "Provider got monthly reward successfully")
 		}
 	}
 
+	var latestSub types.Subscription
+	latestEntryBlock, _, _, found := k.subsFS.FindEntryDetailed(ctx, subObj.Consumer, uint64(ctx.BlockHeight()), &latestSub)
+	if found {
+		latestSub.Credit = latestSub.Credit.SubAmount(totalTokenAmount)
```

Comment:
make sure this is not negative

Reply:
done. see https://github.com/lavanet/lava/pull/1141/commits/7bd5da17e86dc4e5813f3dd539a3aa4bc0d35170